server/aoprotocol.py

```diff
@@ -187,7 +187,8 @@ def net_cmd_hi(self, args):
                 self.client.disconnect()
                 return
 
-        logger.log_server('Connected. HDID: {}.'.format(self.client.hdid), self.client)
+        if self.client.hdid != 'ms2-prober' and self.server.config['show_ms2-prober'] == True:
```

Comment:
This won't display any ms2-prober connection messages regardless of show_ms2-prober being True or False.
Moreover, it won't show any connection messages at all if show_ms2-prober is False regardless of HDID.